treeherder/webapp/api/auth.py

```diff
@@ -16,3 +28,39 @@ def hawk_lookup(id):
         'key': str(credentials.secret),
         'algorithm': 'sha256'
     }
+
+
+class TaskclusterAuthViewSet(viewsets.ViewSet):
+
+    @list_route()
+    def login(self, request):
+        """
+        Verify credentials with Taskcluster
+        """
+        authorization = request.META.get("HTTP_OTH", None)
+
+        # when we switch to Django 1.10, use ``get_host`` and ``get_port``
+        hostparts = request.META["HTTP_HOST"].split(":")
+        host = hostparts[0]
+        if "HTTP_X_FORWARDED_PORT" in request.META:
```

Comment:
ok, cool.  I did this.  It worked fine without changing anything in varnish both on stage and locally, so I guess we're ok there.  Had to add ``USE_X_FORWARDED_PORT = True`` to settings, though.